keps/sig-apps/4017-pod-index-label/README.md

```diff
@@ -0,0 +1,915 @@
+This section must be completed when targeting alpha to a release.
+-->
+The feature can be safely rolled back.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: PodIndexLabel
+  - Components depending on the feature gate:
+    - StatefulSet controller
+    - Job controller
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+Yes - when we start setting a new label, if someone is doing deep-equal comparison, those will start failing.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+Yes. If the feature gate is disabled, the StatefulSet/Job controller will not add the
+pod index as a label.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The StatefulSet/Job controller will begin adding the pod index as a label to
+pods created while the feature is enabled, and existing pods will be unaffected.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+We plan to add unit and integration tests. 
```

Comment:
FWIW - enablement/disablement tests are not tests that are exercising the feature. They rather test "the transition of the feature-gate" [so they switch the value of the FG in the middle of the test].

Given the nature of this feature (it only adds a label for *newly created* pods, enablement/disablement tests don't much value.

So based on the comment above, I suggest changing to:

Given that this feature doesn't introduce any new API field, enablement/disablement tests will not provide reasonable value and won't be added.

Reply:
done